automerge-backend/src/backend.rs

```diff
@@ -230,6 +233,47 @@ impl Backend {
     }
 
     pub fn get_changes(&self, have_deps: &[amp::ChangeHash]) -> Vec<&Change> {
+        if have_deps.len() == 0 {
+            return self
+                .history
+                .iter()
+                .filter_map(|h| self.hashes.get(h))
+                .collect();
+        }
+
+        let lowest_idx = have_deps
+            .iter()
+            .filter_map(|h| self.history_idx.get(h))
+            .min();
+        let lowest_idx = match lowest_idx {
+            Some(i) => i,
+            None => {
+                return vec![];
+            }
+        };
+
+        let mut missing_changes = vec![];
+        let mut has_seen: HashSet<_> = have_deps.iter().collect();
+        for hash in &self.history[*lowest_idx..] {
+            let change = self.hashes.get(hash).unwrap();
+            let mut deps_seen = 0;
+            for dep in &change.deps {
+                if has_seen.contains(dep) {
+                    deps_seen += 1;
+                }
+            }
+            if deps_seen > 0 && deps_seen == change.deps.len() {
+                missing_changes.push(change);
+                has_seen.insert(hash);
+            } else if deps_seen > 0 {
+                break;
+            }
```

Comment:
Could reduce duplication of the `deps_seen > 0` check here by moving the else if to just an else inside the first block.

Reply:
goot catch